logging/runtime/src/main/java/io/quarkiverse/googlecloudservices/logging/runtime/LoggingConfiguration.java

```diff
@@ -0,0 +1,220 @@
+    public static class GcpTracingConfig {
+
+        /**
+         * Use this setting to determine if extracted trace ID's should
+         * also be forwarded to GCP for linking with GCP Operations Tracing.
+         */
+        @ConfigItem(defaultValue = "true")
+        public boolean enabled;
+
+        /**
+         * If the GCP Operations Tracing is in another project, configure it
+         * here. By default the logging project will be used.
+         */
+        @ConfigItem
+        public Optional<String> projectId;
+    }
+
+    @ConfigGroup
+    public static class ResourceConfig {
+
+        /**
+         * The resource type of the log.
+         */
+        @ConfigItem(defaultValue = "global")
+        public String type;
+
+        /**
+         * Resource labels.
+         */
+        @ConfigItem
+        public Map<String, String> label;
+
+    }
+
+    @ConfigGroup
+    public static class MDCConfig {
+
+        /**
+         * Include MDC values in the log.
+         */
+        @ConfigItem(defaultValue = "true")
+        public boolean included;
+
+        /**
+         * Field name for MDC values, defaults to 'mdc'.
+         */
+        @ConfigItem(defaultValue = "mdc")
+        public String fieldName;
+
+    }
+
+    @ConfigGroup
+    public static class StackTraceConfig {
+
+        /**
+         * Include stack traces when exceptions are thrown.
+         */
+        @ConfigItem(defaultValue = "true")
+        public boolean included;
+
+        /**
+         * Should stack traces be rendered as strings or arrays?
+         */
+        @ConfigItem(defaultValue = "STRING")
+        public StackTraceRendering rendering;
+
+        /**
+         * If the stack is rendered as an array, what format should each
+         * stack frame have?
+         */
+        @ConfigItem(defaultValue = "STRING")
+        public StackElementRendering elementRendering;
+
+    }
+
+    @ConfigGroup
+    public static class ParametersConfig {
+
+        /**
+         * Include parameter values in the log.
+         */
+        @ConfigItem(defaultValue = "true")
+        public boolean included;
+
+        /**
+         * Field name for parameter values, defaults to 'parameters'.
+         */
+        @ConfigItem(defaultValue = "parameters")
+        public String fieldName;
+
+    }
+
+    public enum ConfigLevel {
+        DEBUG(Severity.DEBUG),
+        INFO(Severity.INFO),
+        WARN(Severity.WARNING),
+        ERROR(Severity.ERROR),
+        FATAL(Severity.CRITICAL);
+
+        private Severity severity;
+
+        private ConfigLevel(Severity severity) {
+            this.severity = severity;
+        }
+
+        public Severity getSeverity() {
+            return severity;
+        }
+    }
+
+    public enum StackTraceRendering {
+        STRING,
+        ARRAY
+    }
+
+    public enum StackElementRendering {
+        STRING,
+        OBJECT
+    }
+
+    public enum LogFormat {
+        TEXT,
+        JSON
+    }
```

Comment:
I think you can use these enums inside the config class itself, you can check on the core logging extension for example

Reply:
Not entirely sure what you mean. They are used in the config class.

Reply:
Yes, my bad